src/plugins/discover/public/components/discover_grid_flyout/discover_grid_flyout.tsx

```diff
@@ -42,7 +42,19 @@ export interface DiscoverGridFlyoutProps {
 /**
  * Flyout displaying an expanded Elasticsearch document
  */
-export function DiscoverGridFlyout({
+export function DiscoverGridFlyout(props: DiscoverGridFlyoutProps) {
+  const getRenderDocViewerFlyout = useProfileAccessor('getRenderDocViewerFlyout', {
+    record: props.hit,
+  });
```

Comment:
Thanks @davismcphee  for creating this extension point. Althought this extension point works to display our flyout. I am still working on that. What it does not do is gives any handle of `onClose` event to the users of extension point.

For example, consider 2 use cases: 

1. Users click `close` button of the flyout, in that case we want to `effect` the expand icon change in the Discover DataGrid as seen below:

2. User clicking on `expand` button toggles the flyout but the extension point does not have access to know when user has clicked expand icon.

I think giving access to `setExpandedDoc` should be good enough and may also be more powerful for than replacing `DiscoverGridFlytout`.

What do you think?